tests/dialects/test_memref.py

```diff
@@ -293,3 +297,61 @@ def test_memref_cast():
 
     assert cast.source is memref_ssa_value
     assert cast.dest.typ is res_type
+
+
+def test_dma_start():
+    src_type = MemRefType.from_element_type_and_shape(i64, [4, 512], memory_space=1)
+    dest_type = MemRefType.from_element_type_and_shape(i64, [4 * 512], memory_space=2)
+
+    tag_type = MemRefType.from_element_type_and_shape(i32, [4])
+
+    src = TestSSAValue(src_type)
+    dest = TestSSAValue(dest_type)
+    tag = TestSSAValue(tag_type)
+
+    index = TestSSAValue(IndexType())
+    num_elements = TestSSAValue(IndexType())
+
+    dma_start = DmaStartOp.get(
+        src, [index, index], dest, [index], num_elements, tag, [index]
+    )
+
+    dma_start.verify()
+
+    # check that src index count is verified
+    with pytest.raises(VerifyException):
```

Comment:
(Just to mention it) You could check the error message here.

Reply:
Done! You can actually pass `match="..."` to the `pytest.raises` thingy to match, which makes it really nice!